app/src/main/java/com/example/samplegallery/PhotoBlowupFragment.java

```diff
@@ -86,16 +90,32 @@ public void onResponse(JSONObject response) {
                         try {
                             sizes = response.getJSONObject("sizes").getJSONArray("size");
                             photoUrl = sizes
-                                    .getJSONObject(sizes.length() - 1)
+                                    .getJSONObject(sizes.length() - 1) //finding a smaller size will speed up load times
                                     .getString("source");
                         } catch (JSONException e) {
                             e.printStackTrace();
                         }
-
-                        // finally set the url for the image
-                        img.setImageUrl(
-                                photoUrl,
-                                VolleyRequestQueue.getInstance(getContext()).getImgLoader());
+                        //isnt onResponse on the uithread? would moving it off the main thread speed things up? would it be possible to?
```

Comment:
Yeah, it's on the ui thread but only after the response is fetched. That's necessary though since you're updating the ui! The request itself in not on the ui thread